### PR TITLE
win_iis_webapplication: Parameter fixes and check-mode

### PR DESCRIPTION
**Pull request by dagwieers**
##### SUMMARY
This PR includes:
- General cleanup of parameters
- Added check-mode support
- Cleanup of return values

##### ISSUE TYPE

 - Bugfix Pull Request
 - Docs Pull Request

##### COMPONENT NAME

win_iis_webapplication

##### ANSIBLE VERSION

v2.4

### CODE REVIEW

File: lib/ansible/modules/windows/win_iis_webapplication.ps1 [hunk 1]
```diff
@@ -20,100 +20,88 @@
...
+    }
     $result.changed = $true
-
   }
 
   # Remove application
   if ($state -eq 'absent' -and $application) {
-    $application = Remove-WebApplication -Site $site -Name $name
+    if (-not $check_mode) {
+        $application = Remove-WebApplication -Site $site -Name $name
```
Comment (jborean93) [inline lib/ansible/modules/windows/win_iis_webapplication.ps1]: Remove-WebApplication supports `-WhatIf`

**Review by jborean93 — CHANGES_REQUESTED** [review]
Only minor point around removing an application and check mode

File: lib/ansible/modules/windows/win_iis_webapplication.ps1 (hunk 1 above)
Comment (dagwieers) [reply lib/ansible/modules/windows/win_iis_webapplication.ps1]: Done !

**Review by jborean93 — APPROVED** [review]

**Comment by jborean93** [comment]
looks good, will merge through

File: lib/ansible/modules/windows/win_iis_webapplication.ps1 [hunk 2]
```diff
@@ -20,100 +20,82 @@
...
+        $application = New-WebApplication @application_parameters -Force
+    }
     $result.changed = $true
-
   }
 
   # Remove application
   if ($state -eq 'absent' -and $application) {
-    $application = Remove-WebApplication -Site $site -Name $name
+    $application = Remove-WebApplication -Site $site -Name $name $WhatIf:$check_mode
```
Comment (sm4rk0) [inline lib/ansible/modules/windows/win_iis_webapplication.ps1:74]: `$WhatIf:$check_mode` should be `-WhatIf:$check_mode`

Comment (sm4rk0) [reply lib/ansible/modules/windows/win_iis_webapplication.ps1:74]: BTW, is it OK to correct small typos this way, or should I report an issue and/or make a PR?

Comment (jhawkesworth) [reply lib/ansible/modules/windows/win_iis_webapplication.ps1:74]: @sm4rk0 - if a PR has been merged like this one, best to make a new PR - maintainers don't tend to get notified once a PR has been merged.

**Pull request merged**